src/sender.js

```diff
@@ -17,5 +26,14 @@ exports.handler = async (event) => {
 
   const requests = event.Records.map(r => r.body);
 
-  await dispenseTokens(requests, provider, faucetAddr, privKey, amount);
+  // todo, order requests by color here
+  const colorList = requests.reduce(groupValuesByColor, {});
+
+  for (var color in colorList) {
```

Comment:
as a suggestion (your code is also fine)

```
Object.keys(colorList).forEach(color => 
   dispenseTokens(colorList[color], provider, faucetAddr, privKey, amount, parseInt(color))
)
```

more concise + no await on `dispenseTokens` as the UTXO sets for different colors aren't intersecting and no double spend possible